File: plugins/coding/src/syntax/languages/typescript/fixtures/basic.ts
import { readFile } from "fs";
import type { Config } from "./types";

const MAX_RETRIES = 3;

export function greet(name: string): string {
    return `Hello, ${name}!`;
}

function helper(): boolean {
    return true;
}

export interface Processor {
    run(input: string): string;
    reset(): void;
}

export class AppConfig {
    name: string;

    constructor(name: string) {
        this.name = name;
    }

    validate(): boolean {
        return this.name.length > 0;
    }

    reset(): void {
        this.name = "";
    }
}

export enum Status {
    Active = "active",
    Inactive = "inactive",
}

export type Result<T> = { ok: true; value: T } | { ok: false; error: string };

File: plugins/coding/src/syntax/languages/rust/fixtures/basic.rs
use std::collections::HashMap;
use std::io;

const MAX_SIZE: usize = 1024;

pub fn process(input: &str) -> String {
    input.to_uppercase()
}

fn helper() -> bool {
    true
}

pub struct Config {
    name: String,
    value: u32,
}

pub enum Status {
    Active,
    Inactive,
}

pub trait Processor {
    fn run(&self, input: &str) -> String;
    fn reset(&mut self);
}

impl Processor for Config {
    fn run(&self, input: &str) -> String {
        format!("{}: {}", self.name, input)
    }

    fn reset(&mut self) {
        self.value = 0;
    }
}

impl Config {
    pub fn new(name: String) -> Self {
        Self { name, value: 0 }
    }
}
